### PR TITLE
Lesson 03

### PR DESCRIPTION
**Pull request by AlekseiKanash**

### CODE REVIEW

File: lesson_03/strings/strings.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package strings
+
+import "fmt"
+
+func Max(inStrings []string) string {
```
Comment (AlekseiKanash) [inline lesson_03/strings/strings.go]: I wasn't focused on naming and CamelCase - I see no reason. No real use cases - so no design. But I remember that Max would be accessible from other packages and max would not.

File: lesson_03/average/average.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+type ReturnCode int8
+
+const (
+	Success ReturnCode = iota
+	Error
+)
+
+func isNumericKind(kind reflect.Kind) bool {
```
Comment (AlekseiKanash) [inline lesson_03/average/average.go:15]: Maybe there is a standard way to do the same? I did not find.

File: lesson_03/strings/strings.go (hunk 1 above)
Comment (nikonok) [reply lesson_03/strings/strings.go]: Yeah, better choose `max`, cause you use it only from package(with no cases in other packages).

File: lesson_03/average/average.go (hunk 2 above)
Comment (nikonok) [reply lesson_03/average/average.go:15]: I guess that here is two ways `ParseFloat` or the way you do. I don't know any such things like `std::is_integral`=)

File: lesson_03/maps/maps.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+func sortData(data map[int]string) []string {
+	out := []string{}
+	keys := []int{}
+	for k := range data {
+		keys = append(keys, k)
+	}
+
+	sort.Slice(keys, func(i, j int) bool {
```
Comment (nikonok) [inline lesson_03/maps/maps.go]: Maybe you should use [`sort.Ints`](https://pkg.go.dev/sort#Ints) here, but this case works too.

File: lesson_03/main.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package main
+
+import (
+	"github.com/AlekseiKanash/golang-course/lesson_03/average"
+	"github.com/AlekseiKanash/golang-course/lesson_03/maps"
+	"github.com/AlekseiKanash/golang-course/lesson_03/strings"
+)
+
+func main() {
+	average.TestSlice()
```
Comment (nikonok) [inline lesson_03/main.go:10]: I guess it will be useful for you https://pkg.go.dev/testing. We will study it later, but you can use now=)

**Review by nikonok — APPROVED** [review]

**Review by denislituev — APPROVED** [review]
Good job! maybe it seems too complex for this simple tasks... but it works!

File: lesson_03/main.go (hunk 4 above)
Comment (AlekseiKanash) [reply lesson_03/main.go:10]: I would prefer to wait until we will study it, just have not enough time right now.

File: lesson_03/maps/maps.go (hunk 3 above)
Comment (AlekseiKanash) [reply lesson_03/maps/maps.go]: fixed

File: lesson_03/strings/strings.go (hunk 1 above)
Comment (AlekseiKanash) [reply lesson_03/strings/strings.go]: fixed

**Pull request merged**